src/main/java/io/vertx/cassandra/PreparedStatement.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2018 The Vert.x Community.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.vertx.cassandra;
+
+import io.vertx.codegen.annotations.VertxGen;
+import io.vertx.core.Handler;
+
+/**
+ * Analogy of {@link com.datastax.driver.core.PreparedStatement}
+ *
+ * @see com.datastax.driver.core.PreparedStatement
+ */
+@VertxGen
+public interface PreparedStatement {
+
+  /**
+   * Creates a new {@link Statement} object and bind its variables to the  provided values.
+   *
+   * @param params the values to bind
+   * @return instance of {@link Statement} which can be executed with {@link CassandraClient#execute(Statement, Handler)}
+   * @see com.datastax.driver.core.PreparedStatement#bind(Object...)
+   */
+  Statement bind(BindArray params);
```

Comment:
Actually, i tried List<Object> as well, by codegen said this:
```
[ERROR] diagnostic: /Users/sammers/vertx-cassandra-client/src/main/java/io/vertx/cassandra/PreparedStatement.java:38: error: Could not generate model for io.vertx.cassandra.PreparedStatement#bind(java.util.List<java.lang.Object>): type java.util.List<java.lang.Object> is not legal for use for a parameter in code generation
  Statement bind(List<Object> params);
            ^
```

Reply:
Thats why i have handly wrapped List of objects with `BindArray`

Reply:
Indeed, my bad. Then, instead of creating a `BindArray`, how about sticking close to the driver model with a `bind()` method returning a `BoundStatement`?

Reply:
I am not sure about it, the current API is not limited and allow any types. In fact, the option with `bind()` will be limited, since for `Row` we have a lot of `@GenIgnore` methods. In my opinion, it would be better to think about it when we 3.6.0 with new allowed by codegen types will be released.

Reply:
What I was thinking about is something along these lines:

```java
interface PreparedStatement {
  BoundStatement bind();
}

interface BoundStatement extends Statement {
  void set(int pos, Object val);
  void set(String name, Object val);
}
```

Reply:
Ah, I got. Yes, that is fine for me.

Reply:
There is a problem, the driver does not allow the operations:
```
void set(int pos, Object val);
void set(String name, Object val);
```

because BoundStatement has only the following methods for setting:
```
<V> T set(int i, V v, Class<V> targetClass);
<V> T set(int i, V v, TypeToken<V> targetType);
<V> T set(String i, V v, TypeCodec<V> codec);

<V> T set(int name, V v, Class<V> targetClass);
<V> T set(int name, V v, TypeToken<V> targetType);
<V> T set(String name, V v, TypeCodec<V> codec);
```

so we need to have the type information for setters

Reply:
I think you can get it from the `CodecRegistry`. For now I think it's fine to use `CodecRegistry.DEFAULT_INSTANCE`

Reply:
Done